plugins/woocommerce/src/Internal/Fulfillments/Providers/USPSShippingProvider.php

```diff
@@ -3,43 +3,152 @@
+	 * Gets the display name of the provider.
 	 *
-	 * @return string
+	 * @return string The provider name 'USPS'.
 	 */
 	public function get_name(): string {
 		return 'USPS';
 	}
 
 	/**
-	 * Get the icon of the shipping provider.
+	 * Gets the path to the provider's icon.
 	 *
-	 * @return string
+	 * @return string URL to the USPS logo image.
 	 */
 	public function get_icon(): string {
 		return esc_url( WC()->plugin_url() ) . '/assets/images/shipping_providers/usps.png';
 	}
 
 	/**
-	 * Get the tracking URL for a given tracking number.
+	 * Generates the tracking URL for a given tracking number.
 	 *
-	 * @param string $tracking_number The tracking number.
-	 * @return string The tracking URL.
+	 * @param string $tracking_number The tracking number to generate URL for.
+	 * @return string The complete tracking URL.
 	 */
 	public function get_tracking_url( string $tracking_number ): string {
-		return 'https://tools.usps.com/go/TrackConfirmAction?qtc_tLabels1=' . rawurlencode( $tracking_number );
+		return 'https://tools.usps.com/go/TrackConfirmAction?tLabels=' . rawurlencode( $tracking_number );
+	}
+
+	/**
+	 * Gets the list of origin countries supported by USPS.
+	 *
+	 * @return array<string> Array of country codes (only 'US').
+	 */
+	public function get_shipping_from_countries(): array {
+		return array( 'US' ); // USPS only ships from the United States.
+	}
+
+	/**
+	 * Gets the list of destination countries supported by USPS.
+	 *
+	 * @return array<string> Array of country codes including domestic and international.
+	 */
+	public function get_shipping_to_countries(): array {
+		return array_merge(
+			$this->domestic_countries,
+			explode( ' ', 'AD AE AF AG AI AL AM AO AR AT AU AW AZ BA BB BD BE BF BG BH BI BJ BM BN BO BR BS BT BW BY BZ CA CD CF CG CH CI CL CM CN CO CR CU CV CY CZ DE DJ DK DM DO DZ EC EE EG ER ES ET FI FJ FR GA GB GD GE GH GI GM GN GQ GR GT GW GY HK HN HR HT HU ID IE IL IN IQ IR IS IT JM JO JP KE KG KH KI KM KN KP KR KW KZ LA LB LC LK LR LS LT LU LV LY MA MC MD ME MG MK ML MM MN MO MR MT MU MV MW MX MY MZ NA NE NG NI NL NO NP NZ OM PA PE PG PH PK PL PT PY QA RO RS RU RW SA SB SC SD SE SG SI SK SL SM SN SO SR ST SV SY SZ TD TG TH TJ TL TM TN TO TR TT TV TW TZ UA UG UK UY UZ VC VE VN VU WS YE ZA ZM ZW' )
+		);
+	}
+
+	/**
+	 * Attempts to parse and validate a USPS tracking number.
+	 *
+	 * @param string $tracking_number The tracking number to validate.
+	 * @param string $shipping_from Origin country code.
+	 * @param string $shipping_to Destination country code.
+	 * @return array|null Array with tracking URL and score, or null if invalid.
+	 */
+	public function try_parse_tracking_number( string $tracking_number, string $shipping_from, string $shipping_to ): ?array {
+		if ( empty( $tracking_number ) || ! $this->can_ship_from_to( $shipping_from, $shipping_to ) ) {
+			return null; // Invalid input or route.
+		}
+
+		$tracking_number = strtoupper( preg_replace( '/\s+/', '', $tracking_number ) );
```

Comment:
what do you think about being more restrictive about replaced chars?

```suggestion
		$tracking_number = preg_replace('/[^A-Z0-9]/', '', strtoupper($tracking_number));
```

Reply:
Interesting thought. Thanks for suggesting this, makes sense

Reply:
Addressed it globally in the refactor PR: https://github.com/woocommerce/woocommerce/pull/59214/commits/30a80fd715a8e51e1c9b53df1b5f0893e3584208